assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,312 @@
+          var tagList = [];
+          if (imageStream.status.tags) {
+            imageStream.status.tags.forEach(function(item){ 
+              tagList.push(item["tag"]); 
+            });
+          }          
+          $scope.pushTo.imageStreams.push(name);
+          $scope.pushTo.tags[name] = tagList;
+        });
+        if (selectFirstOption) {
+          $scope.options.pickedPushToImageStream = $scope.pushTo.imageStreams[0];
+          $scope.clearSelectedOutputTag();
+        }
+      });
+    }
+
+    $scope.clearSelectedOutputTag = function() {
+      var tags = $scope.pushTo.tags[$scope.options.pickedPushToImageStream];
+      $scope.options.pickedPushToImageStreamTag = _.find(tags, function(tag) { return tag == "latest" }) || tags[0] || "";
+    }
+
+    $scope.save = function() {
+      $scope.disableInputs = true;
+      // Update Configuration
+      $filter('buildStrategy')($scope.updatedBuildConfig).forcePull = $scope.options.forcePull;
+      if ($scope.updatedBuildConfig.spec.strategy.type === "Docker") {
+        $filter('buildStrategy')($scope.updatedBuildConfig).noCache = $scope.options.noCache;
+      }
+
+      var from = {};
+      if ($scope.options.pickedBuildFromType === "ImageStreamTag") {
+        from = {
+          kind: $scope.options.pickedBuildFromType,
+          namespace: $scope.options.pickedBuildFromNamespace,
+          name: $scope.options.pickedBuildFromImageStream + ":" + $scope.options.pickedBuildFromImageStreamTag
+        };
+      } else if ($scope.options.pickedBuildFromType === "DockerImage") {
+        from = {
+          kind: $scope.options.pickedBuildFromType,
+          name: $scope.options.pickedBuildFromDockerImage
+        };
+      }
+      $filter('buildStrategy')($scope.updatedBuildConfig).from = from;
+
+      var to = {
+        kind: "ImageStreamTag",
+        namespace: $scope.options.pickedPushToNamespace,
+        name: $scope.options.pickedPushToImageStream + ":" + $scope.options.pickedPushToImageStreamTag
+      };
+      $scope.updatedBuildConfig.spec.output.to = to;
+
+      // Update envVars
+      var updateEnvVars = [];
+      angular.forEach($scope.envVars, function(v, k) {
+        var env = {
+          name: k,
+          value: v
+        };
+        updateEnvVars.push(env);
+      });
+      $filter('buildStrategy')($scope.updatedBuildConfig).env = updateEnvVars
+
+      // Update triggers
+      var triggers = [];
+      if ($scope.triggers.webhook) {
+        var webhook = _.find($scope.buildConfig.spec.triggers, function(obj) { return obj.type == "GitHub" })
+        if (!webhook) {
+          webhook =  {
+            github: {
+              secret: ApplicationGenerator._generateSecret()
+            },
+            type: "GitHub"
+          };
+        }
+        triggers.push(webhook);
+      }
+      triggers.push(_.find($scope.buildConfig.spec.triggers, function(obj) { return obj.type == "Generic" }));
```

Comment:
We might change these calls to `_.find` to use `_.filter` in case there is more than one trigger with the same type. (Sorry if that's not what I suggested before.)

```
triggers.concat(_.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type == "Generic" }));
```